website/blog/2024/09-30-jsdom-based-scraping/index.md

```diff
@@ -0,0 +1,225 @@
+---
+slug: scrape-using-jsdom
+title: 'Optimizing web scraping: Transitioning from browser-based to JSDOM'
+description: 'Learn how to scrape using JSDOM, alternative to Cheerio and browser based scraping.'
+image: ./img/jsdom.webp
+author: Saurav Jain
+authorTitle: Developer Community Manager
+authorURL: https://github.com/souravjain540
+authorImageURL: https://avatars.githubusercontent.com/u/53312820?v=4&s=48
+authorTwitter: sauain
+---
+
+# Optimizing web scraping: Transitioning from browser-based to JSDOM
+
+As scraping developers, we sometimes need to extract authentication data like temporary keys to perform our tasks. However, it is not as simple as that. Usually, it is in HTML or XHR network, but sometimes, the auth data is computed. In that case, we either can reverse engineer the computation, which takes a lot of time deobfcusating scripts or run the JavaScript that computes it. Normally, we use a browser, but tsat is expensive. Crawlee provides support for running browser scraper and Cheerio Scraper in parallel, but that is very complex and expensive in terms of compute resource usage. JSDOM helps us run page JavaScript with fewer resources than a browser and slightly higher than Cheerio.
+
+This article will discuss a new approach that we use in one of our Actors to obtain the authentication data from TikTok ads creative centre generated by browser web applications without actually running the browser but instead of it, using JSDOM.
+
+![JSDOM based approach from scraping](./img/jsdom.webp)
+
+<!--truncate-->
+
+## Analysing the website
```

Comment:
```suggestion
## Analyzing the website
```

US English

Reply:
Thanks, I will keep it in mind next time.